src/pprof.c

```diff
@@ -520,6 +520,8 @@ DProf *pprof_Init(DProf *dp, const char **sample_names,
     if (!dp)
       return NULL;
     dp->ownership = 1;
+  } else {
+    dp->ownership = 0;
```

Comment:
Right, this is a good point.  I'll have to update the use of this pattern elsewhere--thanks!